src/strawberry_sqlalchemy_mapper/field.py

```diff
@@ -0,0 +1,731 @@
+                                    for nodes in asyncgen_nodes.values()
+                                ),
+                            ),
+                        )
+                    )
+
+                    # Resolve any generator to lists
+                    resolved = {
+                        node_t: list(nodes) for node_t, nodes in resolved.items()
+                    }
+                    return [
+                        resolved[index_map[gid][0]][index_map[gid][1]] for gid in ids
+                    ]
+
+                return resolve()
+
+            # Resolve any generator to lists
+            resolved = {
+                node_t: list(cast(Iterator[relay.Node], nodes))
+                for node_t, nodes in resolved_nodes.items()
+            }
+            return [resolved[index_map[gid][0]][index_map[gid][1]] for gid in ids]
+
+        return resolver
+
+
+class StrawberrySQLAlchemyConnectionExtension(relay.ConnectionExtension):
+    def apply(self, field: StrawberrySQLAlchemyField) -> None:  # type: ignore[override]
+        from strawberry_sqlalchemy_mapper.mapper import StrawberrySQLAlchemyType
+
+        self.field = field
+        strawberry_definition = get_object_definition(field.type, strict=True)
+        node_type = strawberry_definition.type_var_map.get("NodeType")
+        if node_type is None:
+            specialized_type_var_map = strawberry_definition.specialized_type_var_map
+            node_type = (
+                specialized_type_var_map.get("NodeType")
+                if specialized_type_var_map is not None
+                else None
+            )
+
+        if node_type is None:
+            raise RelayWrongAnnotationError(field.name, cast(type, field.origin))
+
+        assert isinstance(node_type, type)
+        sqlalchemy_definition = StrawberrySQLAlchemyType[Any].from_type(
+            node_type,
+            strict=True,
+        )
+        model = sqlalchemy_definition.model
+        args: dict[str, StrawberryArgument] = {
+            a.python_name: a for a in field.arguments
+        }
+        field.arguments = list(args.values())
+
+        if field.base_resolver is None:
+            if (field_sessionmaker := field.sessionmaker) is None:
+                raise TypeError(
+                    f"Missing `sessionmaker` argument for field {field.name}"
+                )
+
+            def default_resolver(
+                root: Optional[Any],
+                info: Info,
+                **kwargs: Any,
+            ) -> Iterable[Any]:
+                session = connection_session.get()
+                if session is None:
+                    session = field_sessionmaker()
+
+                def _get_query(s: Session):
+                    if root is not None:
+                        # root won't be None when resolving nested connections.
+                        # TODO: Maybe we want to send this to a dataloader?
+                        query = getattr(root, field.python_name)
```

Comment:
I almost want to log a warning here because this will be so bad for performance. But, I think we'd only want to log it once per query, rather than per resolver, which maybe we don't have an easy way to do. So maybe no action, just thinking out loud.